### PR TITLE
Add Scopes into Transaction

### PR DESCRIPTION
**Pull request by shargon**
It's required to have access to the transaction's scopes inside the smart contract in certain cases, I thought about create a new syscall, check if a witness it will be valid, but it depend of the callingscript, executionscript, in fact the whole ExecutionEngine...
I want to check inside my verify if the transaction was signed by the user.

### CODE REVIEW

**Comment by erikzhang** [comment]
How do you decode it in a contract?

**Comment by shargon** [comment]
Is not required to be decoded, you can hardcode the expected one, and don't allow anything else

File: src/neo/Network/P2P/Payloads/Transaction.cs [hunk 1]
```diff
@@ -464,6 +464,7 @@ public StackItem ToStackItem(ReferenceCounter referenceCounter)
                 NetworkFee,
                 ValidUntilBlock,
                 Script,
+                new Array(referenceCounter, Signers.Select(u => new ByteString(u.ToArray())))
```
Comment (Ashuaidehao) [inline src/neo/Network/P2P/Payloads/Transaction.cs:467]: Can you change it to a Map[hash=>scope] ?

Comment (shargon) [reply src/neo/Network/P2P/Payloads/Transaction.cs]: Done

**Comment by erikzhang** [comment]
I think we should create a new syscall, and decode `senders` for the user.

**Comment by shargon** [comment]
> I think we should create a new syscall, and decode senders for the user.

`GetTransactionSigners`?

**Pull request closed without merging**